src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -1484,6 +1484,31 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return False
         End Function
 
+        ' TypeDeclaration  ::=
+        '    ModuleDeclaration  |
+        '    NonModuleDeclaration
+        ' NonModuleDeclaration  ::=
+        '    EnumDeclaration  |
+        '    StructureDeclaration  |
+        '    InterfaceDeclaration  |
+        '    ClassDeclaration  |
+        '    DelegateDeclaration
+        Public Function IsTypeDeclaration(node As SyntaxNode) As Boolean Implements ISyntaxFactsService.IsTypeDeclaration
+            Select Case node.Kind()
+                Case SyntaxKind.EnumStatement,
+                     SyntaxKind.EnumBlock,
+                     SyntaxKind.StructureStatement,
+                     SyntaxKind.StructureBlock,
+                     SyntaxKind.InterfaceStatement,
+                     SyntaxKind.InterfaceBlock,
+                     SyntaxKind.ClassStatement,
+                     SyntaxKind.ClassBlock
```

Comment:
i'm not sure i'm a fan of supporting both the blocks and the statements.  do yo need both?  also: delegates and modules for VB.

Reply:
Probably not, blocks seems correct here